presto-router/src/main/java/com/facebook/presto/router/scheduler/CustomSchedulerManager.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.router.scheduler;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.spi.router.Scheduler;
+import com.facebook.presto.spi.router.SchedulerFactory;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.facebook.presto.util.PropertiesUtil.loadProperties;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public class CustomSchedulerManager
+{
+    private static final Logger log = Logger.get(CustomSchedulerManager.class);
+
+    private static final File CONFIG_FILE = new File("etc/router-config/router-scheduler.properties");
+    private static final String NAME_PROPERTY = "router-scheduler.name";
+
+    private final AtomicBoolean required = new AtomicBoolean();
```

Comment:
If this is meant to guard against errant code calling `getScheduler` before a custom scheduler is loaded, you already guard against this on L77